### PR TITLE
Bug 1123683 - set default testing preferences

### PR DESCRIPTION
**Pull request by galgeek**
Hello, @whimboo and @chmanchester!

Here's a PR setting default testing preferences in firefox_ui_harness/runtest.py.

The preferences set here are based on those set for mozmill tests, here:
http://mxr.mozilla.org/mozilla-beta/source/testing/mozbase/mozprofile/mozprofile/profile.py#330
and here:
https://github.com/mozilla/mozmill/blob/master/mozmill/mozmill/__init__.py#L138

We’ve also referred to the Mochitest preferences set, here:
http://mxr.mozilla.org/mozilla-central/source/testing/profiles/prefs_general.js
along with this WebDriver set:
https://code.google.com/p/selenium/source/browse/javascript/firefox-driver/webdriver.json

Additional default preferences are set by the Marionette client, here:
https://dxr.mozilla.org/mozilla-central/source/testing/marionette/client/marionette/geckoinstance.py#18

Additional notes are available at this etherpad: 
https://etherpad.mozilla.org/automation-firefox-ui-tests

### CODE REVIEW

File: firefox_ui_harness/runtests.py [hunk 1]
```diff
@@ -16,7 +16,46 @@
 
 class ReleaseTestRunner(BaseMarionetteTestRunner):
     extra_prefs = {
-        "browser.tabs.remote.autostart": False,
+        'browser.tabs.remote.autostart': False,
+        'app.update.auto': False,
```
Comment (whimboo) [inline firefox_ui_harness/runtests.py]: Please make sure to sort all the preferences alphabetically. Beside that I would prefer if you could move out this big dictionary from the class into the global scope, or better into a separate file which we can import. Maybe default_prefs.py.

File: firefox_ui_harness/runtests.py [hunk 2]
```diff
@@ -16,7 +16,46 @@
...
+        'browser.startup.page': 0,
+        'browser.tabs.animate': False,
+        'browser.tabs.warnOnClose': False,
+        'browser.tabs.warnOnOpen': False,
+        'browser.uitour.enabled': False,
+        'browser.warnOnQuit': False,
+        'datareporting.healthreport.service.enabled': False,
+        'datareporting.healthreport.uploadEnabled': False,
+        'datareporting.healthreport.documentServerURI': "http://%(server)s/healthreport/",
+        'datareporting.healthreport.about.reportUrl': "http://%(server)s/abouthealthreport/",
```
Comment (whimboo) [inline firefox_ui_harness/runtests.py]: Again, please check if we really need all of them if the service is disabled anyway. You could check via hg check-ins how added those prefs for the mochitest framework.

**Comment by galgeek** [comment]
Hello!

Regarding datareporting.healthreport prefs, I haven't managed to find datareporting.healthreport.service.enabled referenced much, but absence of evidence is not evidence of absence, at least with my limited experience searching at dxr.mozilla.org and mxr.mozilla.org. In the meantime, I suspect the best practice may still be to set documentServerURI and reportURL, as the Mochitest prefs do.

**Comment by whimboo** [comment]
PR has been merged as https://github.com/mozilla/firefox-ui-tests/commit/16a2a3a8dd49c1e5854b2e7cf506b4876b59fa7e

**Pull request closed without merging**